### PR TITLE
Delete plugin selection modes other than `-war`

### LINKED ISSUE #388: Delete plugin selection modes other than `-war`
`jenkinsci/bom` and @cloudbees both run PCT using the `-war` option. To simplify the code base, we should delete all other run modes—checking the update center (including both explicit `-updateCenterUrl` and the default UC), `-parentCoordinates` (whatever that does), `-bom` (#202), probably `-overridenPlugins` _[sic]_ (#88).

As of this writing, `bom` uses

* `-war`
* `-includePlugins`
* `-workDirectory`
* `-reportFile`
* `-m2SettingsFile`
* `-mavenProperties`
* `-excludeHooks org.jenkins.tools.test.hook.TransformPom` (see #387)
* `-mavenPropertiesFile`
* `-skipTestCache`

CloudBees also uses

* `-storeAll`
* `-fallbackGitHubOrganization`
* `-mvn`
* `-hookPrefixes`
* `-externalHooksJars`
* `-testJDKHome`
* `-localCheckoutDir`
* `-includePlugins`

CC @jtnord

### PR DESCRIPTION
**Pull request by basil**
Fixes #388 by building on top of #414. This change deletes most of the remaining tests in this repository, since by now they have become almost completely stale. This isn't really a regression in automated testing as it is an exposure of how little value the existing tests had with regard to current use cases. In a future PR I plan to introduce a new automated test suite that actually covers current use cases and should be easier to maintain as well. I tested this in https://github.com/jenkinsci/bom/pull/1716.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 1]
```diff
@@ -49,18 +47,9 @@
  * @author Frederic Camblor
  */
 public class CliOptions {
-    @Parameter(names = "-updateCenterUrl",
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:52]: Used by neither CloudBees nor `jenkinsci/bom`.

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 2]
```diff
@@ -49,18 +47,9 @@
...
-            description = "Update center JSON file URL")
-    private String updateCenterUrl = null;
-
-    @Parameter(names = "-parentCoordinates",
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:56]: Used by neither CloudBees nor `jenkinsci/bom`.

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 3]
```diff
@@ -153,10 +142,6 @@ public class CliOptions {
     @Parameter(names="-help", description = "Print this help message", help = true)
     private boolean printHelp;
 
-    @Parameter(names = "-overridenPlugins", description = "List of plugins to use to test a plugin in place of the normal dependencies." +
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:156]: Used by neither CloudBees nor `jenkinsci/bom`.

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 4]
```diff
@@ -167,17 +152,6 @@ public class CliOptions {
     )
     private Boolean storeAll = null;
 
-    @Parameter(names = "-bom", description = "BOM file to be used for plugin versions rather than an Update Center or War file")
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:170]: Used by neither CloudBees nor `jenkinsci/bom`.

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 5]
```diff
@@ -285,16 +249,6 @@ public Boolean isStoreAll() {
         return storeAll;
     }
 
-    public static class PluginConverter implements IStringConverter<PCTPlugin> {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:288]: No longer needed as of the removal of `-overridenPlugin`.

File: src/main/java/org/jenkins/tools/test/CliOptions.java [hunk 6]
```diff
@@ -318,16 +272,4 @@ public void validate(String name, String value) throws ParameterException {
             }
         }
     }
-
-    public static class PluginValidator implements IParameterValidator {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/CliOptions.java:322]: No longer needed as of the removal of `-overridenPlugin`.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 7]
```diff
@@ -117,31 +109,6 @@ public PluginCompatTester(PluginCompatTesterConfig config){
         runner = new ExternalMavenRunner(config.getExternalMaven());
     }
 
-    private SortedSet<MavenCoordinates> generateCoreCoordinatesToTest(UpdateSite.Data data, PluginCompatReport previousReport){
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:120]: Was only used in non-WAR modes so therefore deleted.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 8]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
         }
 
         // Determine the plugin data
-        HashMap<String,String> pluginGroupIds = new HashMap<>();  // Used to track real plugin groupIds from WARs
+        Map<String, String> pluginGroupIds = new HashMap<>();  // Used to track real plugin groupIds from WARs
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:127]: General Java code style is to use the interface type on the left hand side of a variable declaration.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 9]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
...
 
         // Scan bundled plugins
         // If there is any bundled plugin, only these plugins will be taken under the consideration for the PCT run
-        UpdateSite.Data data;
-        if (config.getBom() != null) {
-            data = scanBom(pluginGroupIds, "([^/.]+)[.][hj]pi");
-        } else {
-            data = config.getWar() == null ? extractUpdateCenterData(pluginGroupIds) : scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi");
-        }
+        UpdateSite.Data data = scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi");
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:131]: Simplifying the code to only handle the WAR case.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 10]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
...
         if (!data.plugins.isEmpty()) {
             // Scan detached plugins to recover proper Group IDs for them
             // At the moment, we are considering that bomfile contains the info about the detached ones
-            UpdateSite.Data detachedData = config.getBom() != null ? null : config.getWar() != null ? scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:detached-)?plugins/([^/.]+)[.][hj]pi") : extractUpdateCenterData(pluginGroupIds);
+            UpdateSite.Data detachedData = scanWAR(config.getWar(), pluginGroupIds, "WEB-INF/(?:detached-)?plugins/([^/.]+)[.][hj]pi");
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:135]: Simplifying the code to only handle the WAR case.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 11]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
...
 
             // Add detached if and only if no added as normal one
-            UpdateSite.Data finalData = data;
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:176]: Removing pointless layer of indirection.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 12]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
...
-            for (String plugin : pluginsToInclude) {
-                UpdateSite.Plugin pluginData = ucData.plugins.get(plugin);
-                if (pluginData != null) {
-                    LOGGER.log(Level.INFO, "Adding {0} to the test scope", plugin);
-                    pluginsToCheck.put(plugin, pluginData);
-                }
-            }
-        } else {
-            pluginsToCheck = data.plugins;
-        }
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:202]: Simplifying the code to only handle the WAR case.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 13]
```diff
@@ -157,61 +124,37 @@ public PluginCompatReport testPlugins()
...
-
-        if (pluginsToCheck.isEmpty()) {
+        if (data.plugins.isEmpty()) {
             throw new IOException("List of plugins to check is empty, it is not possible to run PCT");
         }
 
-        // if there is only one plugin and it's not already resolved (not in the war, not in a bom and not in an update center)
+        // if there is only one plugin and it's not already resolved (not in the war)
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:151]: Simplifying the code to only handle the WAR case.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 14]
```diff
@@ -220,17 +163,16 @@ public PluginCompatReport testPlugins()
 
         PluginCompatReport report = PluginCompatReport.fromXml(config.reportFile);
 
-        SortedSet<MavenCoordinates> testedCores = config.getWar() == null ? generateCoreCoordinatesToTest(data, report) : coreVersionFromWAR(data);
+        MavenCoordinates coreCoordinates = new MavenCoordinates("org.jenkins-ci.main", "jenkins-war", data.core.version);
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:166]: Simplifying the code to only handle the WAR case. While I was here I also updated the group ID and artifact ID to match the actual ones of the Jenkins WAR. The previous ones were incorrect. This doesn't affect the logic, but is only used for display purposes during logging. Using the correct group ID and artifact ID makes for a less confusing experience when reading the logs.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 15]
```diff
@@ -220,17 +163,16 @@ public PluginCompatReport testPlugins()
...
 
         MavenRunner.Config mconfig = new MavenRunner.Config(config);
         // TODO REMOVE
         mconfig.userProperties.put( "failIfNoTests", "false" );
         report.setTestJavaVersion(config.getTestJavaVersion());
 
         boolean failed = false;
-        ROOT_CYCLE: for(MavenCoordinates coreCoordinates : testedCores){
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:231]: Simplifying the code to only handle the WAR case.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 16]
```diff
@@ -906,6 +683,10 @@ private UpdateSite.Data scanWAR(File war, @NonNull Map<String, String> pluginGro
                                 shortName = m.group(1);
                             }
                         }
+                        String longName = manifest.getMainAttributes().getValue("Long-Name");
+                        if (longName != null) {
+                            plugin.put("title", longName);
+                        }
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:689]: Unrelated bug fix while I was here: the non-WAR modes correctly parsed the title, but WAR mode never did. This doesn't affect production logic, but it is used for display purposes. Using a proper title makes it easier to read the logs.

File: src/main/java/org/jenkins/tools/test/hook/MinaSshdApi.java [hunk 17]
```diff
@@ -16,11 +16,6 @@ protected String getParentProjectName() {
         return "mina-sshd-api-parent";
     }
 
-    @Override
-    protected String getPluginFolderName(UpdateSite.Plugin currentPlugin){
-        return currentPlugin.getDisplayName();
-    }
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/MinaSshdApi.java:22]: Small bug fix: this should never have been overridden in the first place (my bad!) but happened to work as long as we were using the short name as the display name. Now that we fixed the other bug, we have to fix this bug as well.

File: src/main/java/org/jenkins/tools/test/model/PluginRemoting.java [hunk 18]
```diff
@@ -118,13 +118,7 @@ private String retrievePomContentFromInputStream(InputStream pluginUrlStream) th
                 zipEntry = zin.getNextEntry();
             }
 
-            StringBuilder sb = new StringBuilder();
-            byte[] buf = new byte[1024];
-            int n;
-            while ((n = zin.read(buf, 0, 1024)) > -1)
-                sb.append(new String(buf, 0, n, Charset.defaultCharset()));
-
-            return sb.toString();
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/PluginRemoting.java:127]: Unrelated code cleanup while I was here: using Java 11 functionality where possible. This should also be in UTF-8 because this is a POM file built by Maven by the plugin maintainer, not a file written in the current user's default character set.

File: src/test/java/org/jenkins/tools/test/PluginCompatTesterTest.java [hunk 19]
```diff
@@ -87,176 +63,6 @@ public void setUp() throws Exception {
         }
     }
 
-    @Ignore("TODO broken by https://github.com/jenkinsci/active-directory-plugin/releases/tag/active-directory-2.17; figure out how to pin a version")
```
Comment (basil) [inline src/test/java/org/jenkins/tools/test/PluginCompatTesterTest.java:90]: As explained in the PR description, most of these tests were already disabled or testing unused modes. A new test suite is overdue but not handled in this PR.

**Review by imonteroperez — APPROVED** [review]

File: README.md [hunk 20]
```diff
@@ -149,33 +149,6 @@ Currently Java 17 and Java 11 are bundled (JDK_VERSION= {17, 11}).
 
 Full list of options be found [here](./Makefile).
 
-### Running PCT with different version of dependencies
```
[the remark this answers is not in the document]
Comment (jtnord) [reply README.md:152]: did you mean `overridenPlugins` not `includePlugins`?

`includePlugins` is still in the source https://github.com/jenkinsci/plugin-compat-tester/pull/419/files#diff-974ab2995b72e238f0b5447579e1ecd69b65f87c3970900501378daac90323ccR72-R75 and is used to split testing of plugins in the war so that testing can be paralleized

**Review by jtnord — COMMENTED** [review]
looks ok pending build feedback

File: README.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (basil) [reply README.md:152]: > did you mean overridenPlugins not includePlugins?

Yes but the error was in my self-review, not in the code, so there is no code change to be made
